### PR TITLE
feat(ecosystem): Open permissions for code mappings to members

### PR DESCRIPTION
**Pull request by scttcper**
Loosens permissions to anyone with `org:read` for code owners, code mappings.

fixes https://getsentry.atlassian.net/browse/WOR-2186

### CODE REVIEW

File: src/sentry/api/bases/organization.py [hunk 1]
```diff
@@ -103,6 +103,15 @@ class OrganizationIntegrationsPermission(OrganizationPermission):
     }
 
 
+class OrganizationIntegrationsLoosePermission(OrganizationPermission):
+    scope_map = {
+        "GET": ["org:read", "org:write", "org:admin", "org:integrations"],
+        "POST": ["org:read", "org:write", "org:admin", "org:integrations"],
+        "PUT": ["org:read", "org:write", "org:admin", "org:integrations"],
+        "DELETE": ["org:read", "org:write", "org:admin", "org:integrations"],
+    }
```
Comment (armenzg) [inline src/sentry/api/bases/organization.py:112]: For posterity, the current change of permissions would look like this:
```diff
         "GET":     ["org:read", "org:write", "org:admin", "org:integrations"],
-        "POST":    [            "org:write", "org:admin", "org:integrations"],
+        "POST":    ["org:read", "org:write", "org:admin", "org:integrations"],
-        "PUT":     [            "org:write", "org:admin", "org:integrations"],
+        "PUT":     ["org:read", "org:write", "org:admin", "org:integrations"],
-        "DELETE":  [                         "org:admin", "org:integrations"],
+        "DELETE":  ["org:read", "org:write", "org:admin", "org:integrations"],
```

https://github.com/getsentry/sentry/blob/c605f45a3ca0e585cd7ca1c5a0c286c0d61913e3/src/sentry/api/bases/organization.py#L97-L103

File: src/sentry/api/endpoints/organization_code_mapping_details.py [hunk 2]
```diff
@@ -5,7 +5,10 @@
 from rest_framework.response import Response
 
 from sentry.api.base import region_silo_endpoint
-from sentry.api.bases.organization import OrganizationEndpoint, OrganizationIntegrationsPermission
```
Comment (armenzg) [inline src/sentry/api/endpoints/organization_code_mapping_details.py:8]: This grants any organization _member_ to read, write, update and delete the _details_ of code mappings. Is this correct?

File: src/sentry/api/endpoints/organization_codeowners_associations.py [hunk 3]
```diff
@@ -9,7 +9,7 @@
 
 
 class OrganizationCodeOwnersAssociationsPermission(OrganizationPermission):
-    scope_map = {"GET": ["org:integrations"]}
+    scope_map = {"GET": ["org:read", "org:write", "org:integrations"]}
```
Comment (armenzg) [inline src/sentry/api/endpoints/organization_codeowners_associations.py:12]: I don't see `org:admin` listed here; does this mean that admins cannot make changes to code owners?

File: src/sentry/api/endpoints/organization_code_mapping_details.py (hunk 2 above)
Comment (scttcper) [reply src/sentry/api/endpoints/organization_code_mapping_details.py:8]: that's the goal i believe

File: src/sentry/api/endpoints/organization_codeowners_associations.py (hunk 3 above)
Comment (scttcper) [reply src/sentry/api/endpoints/organization_codeowners_associations.py:12]: this one was weird, maybe we should re-use the same permissions

File: tests/sentry/api/endpoints/test_organization_code_mapping_details.py [hunk 4]
```diff
@@ -57,6 +68,11 @@ def test_basic_edit(self):
         assert resp.data["id"] == str(self.config.id)
         assert resp.data["sourceRoot"] == "newRoot"
 
+    def test_basic_edit_from_member_permissions(self):
+        self.login_as(user=self.user2)
+        resp = self.make_put({"sourceRoot": "newRoot"})
+        assert resp.status_code == 200
```
Comment (armenzg) [inline tests/sentry/api/endpoints/test_organization_code_mapping_details.py:74]: This is updating an existing code mapping:
`source_root: "/source/root" -> "newRoot"`

This is the code mapping:
https://github.com/getsentry/sentry/blob/c605f45a3ca0e585cd7ca1c5a0c286c0d61913e3/tests/sentry/api/endpoints/test_organization_code_mapping_details.py#L38-L45

File: tests/sentry/api/endpoints/test_organization_code_mappings.py [hunk 5]
```diff
@@ -232,6 +232,11 @@ def test_basic_post_with_valid_integrationId(self):
             "defaultBranch": "master",
         }
 
+    def test_basic_post_from_member_permissions(self):
+        self.login_as(user=self.user2)
+        response = self.make_post()
+        assert response.status_code == 201, response.content
```
Comment (armenzg) [inline tests/sentry/api/endpoints/test_organization_code_mappings.py:238]: This creates a new code mapping by a non-superuser user:
https://github.com/getsentry/sentry/blob/c605f45a3ca0e585cd7ca1c5a0c286c0d61913e3/tests/sentry/api/endpoints/test_organization_code_mappings.py#L15

File: tests/sentry/api/endpoints/test_organization_codeowners_associations.py [hunk 6]
```diff
@@ -105,19 +105,14 @@ def test_response_data_is_correct(self):
             in response.data[self.project_2.slug]["errors"]["teams_without_access"]
         )
 
-    def test_no_access(self):
+    def test_member_can_access(self):
         """
-        Tests that users without the 'org:integrations' scope (i.e. Members) cannot access this endpoint.
+        Tests that users without the 'org:read' scope (i.e. Members) can access this endpoint.
```
Comment (armenzg) [inline tests/sentry/api/endpoints/test_organization_codeowners_associations.py:110]: I would like to know the difference between `org:read` vs `org:write` to feel comfortable about giving `org:read` access to the endpoints.

File: tests/sentry/api/endpoints/test_organization_codeowners_associations.py [hunk 7]
```diff
@@ -105,19 +105,14 @@ def test_response_data_is_correct(self):
...
         """
         member = self.create_user("hernando@life.com")
         self.create_member(user=member, organization=self.organization, role="member")
```
Comment (armenzg) [inline tests/sentry/api/endpoints/test_organization_codeowners_associations.py:113]: The creation of users and members in here are slightly different than the other `setUp` methods I have looked at:
https://github.com/getsentry/sentry/blob/c605f45a3ca0e585cd7ca1c5a0c286c0d61913e3/tests/sentry/api/endpoints/test_organization_code_mappings.py#L15
https://github.com/getsentry/sentry/blob/c605f45a3ca0e585cd7ca1c5a0c286c0d61913e3/tests/sentry/api/endpoints/test_organization_code_mappings.py#L23-L28

This is just an observation. It probably does not matter.

**Review by NisanthanNanthakumar — CHANGES_REQUESTED** [review]
needs a small change

File: src/sentry/api/bases/organization.py (hunk 1 above)
Comment (armenzg) [reply src/sentry/api/bases/organization.py:112]: As per @NisanthanNanthakumar the member role doesn't have org:write here
https://github.com/getsentry/sentry/blob/f6177828e14fe567b69eea60520cae7262eba39e/src/sentry/conf/server.py#L1652-L1666

**Review by armenzg — APPROVED** [review]

**Review by NisanthanNanthakumar — APPROVED** [review]

**Pull request merged**